src/plugins/curator/components/promotion-details-fields/Slots.jsx

```diff
@@ -0,0 +1,216 @@
+import humanizeEnums from '@triniti/cms/utils/humanizeEnums';
+import SelectField from '@triniti/cms/components/select-field';
+import { Button, Col, FormGroup, Icon, ListGroupItem } from '@triniti/admin-ui-plugin/components';
+import WidgetPickerField from '@triniti/cms/plugins/curator/components/widget-picker-field';
+import SlotRendering from '@triniti/schemas/triniti/curator/enums/SlotRendering';
+
+const renderingOptions = humanizeEnums(SlotRendering, { format: 'map', except: [SlotRendering.UNKNOWN] });
+
+export default class Slots extends React.Component {
+  static propTypes = {
+    borderClass: PropTypes.string,
+    fields: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+    readOnly: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    borderClass: null,
+    readOnly: false,
+  };
+
+  constructor(props) {
+    super(props);
+    this.handleAdd = this.handleAdd.bind(this);
+    this.handleMoveDown = this.handleMoveDown.bind(this);
+    this.handleMoveUp = this.handleMoveUp.bind(this);
+  }
+
+  componentDidMount() {
+    this.sortable = Sortable.create(this.sortableListRef, {
+      animation: 150,
+      chosenClass: 'active',
+      draggable: '.sortable-slots',
+      filter: '.js-remove',
+      handle: '.sort-handle',
+      onUpdate: () => {
+        const { fields } = this.props;
+        const newSlotOrder = this.sortable.toArray();
+        fields.removeAll();
+        newSlotOrder.forEach((slot) => {
+          try {
+            fields.push(JSON.parse(slot));
+          } catch (e) {
+            fields.push();
+          }
+        });
+      },
+    });
+  }
+
+  handleAdd() {
+    const { fields } = this.props;
+    fields.push({ name: "" });
+
+    if (fields.length < 1) {
+      return;
+    }
+
+    this.scrollToSlot(this.sortableListRef.scrollHeight + 600);
+  }
+
+  handleMoveDown(index) {
+    const { fields } = this.props;
+    if (index === fields.length - 1) {
+      return;
+    }
+
+    fields.move(index, index + 1);
+  }
+
+  handleMoveUp(index) {
+    const { fields } = this.props;
+    if (index === 0) {
+      return;
+    }
+
+    fields.move(index, index - 1);
+  }
+
+  scrollToSlot(offset) {
+    const screenBody = document.getElementsByClassName('screen-body')[0];
+    screenBody.scrollTo({ top: offset, behavior: 'smooth' });
+  }
+
+  render() {
+    const { fields, readOnly, borderClass } = this.props;
+
+    return (
+      <ul className="list-group list-group-flush" ref={(el) => { this.sortableListRef = el; }}>
+        <ListGroupItem className="p-0 pb-2">
+          {!readOnly && (
+            <Button color="primary" onClick={() => this.handleAdd()}>
+              <Icon imgSrc="plus" alt="+" className="mr-1" />Add a Slot
+            </Button>
+          )}
+        </ListGroupItem>
+
+        {fields.getAll().map((slot, index) => (
+          <ListGroupItem
+            style={{"z-index": String(99 - index)}}
```

Comment:
@albertgcsula - is there a real need for the `z-index` rule here? I'm getting this warning in the devTools:
`Unsupported style property z-index. Did you mean zIndex?`

Reply:
@arunkarnati fixed